sklearn/cluster/_optics_inner.pyx

```diff
@@ -0,0 +1,28 @@
+cimport numpy as np
+import numpy as np
+cimport cython
+
+ctypedef np.float64_t DTYPE_t
+ctypedef np.int_t DTYPE
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cpdef quick_scan(double[:] rdists, double[:] dists):
```

Comment:
It looks like it gets an argmin on `rdists` returning the argmin on `dists` in case of tie.

Is it premature optimisation? Does it provide major gains (for realistic sizes) relative to:
```
d = rdists.min()
mask = (rdists == d)
if mask.sum() == 1:
    return np.argmax(mask)
else:
    ....
```
or relative to just doing a lexsort?

Reply:
Added comment on function.

I remember that it provided pretty major gains, but I haven't compared it your suggestion above, so I'll benchmark to compare and post what I find.